std/range/package.d

```diff
@@ -11844,3 +11848,67 @@ pure @safe unittest
     static immutable r2 = [1, 2, 3, 4, 0, 0];
     assert(r1.padRight(0, 6).equal(r2));
 }
+
+private struct TakeWhile(alias pred, R)
+if (isInputRange!R && is(typeof(!unaryFun!pred(ElementType!R.init)) == bool))
+{
+    private R _range;
+
+    // Range primitives
+    bool empty() @property
+    {
+        return _range.empty || !(unaryFun!pred(_range.front));
+    }
+
+    auto front() @property
+    {
+        assert(!empty, "Attempting to fetch the front of an empty TakeWhile range");
+        return _range.front;
+    }
+
+    auto save() @property
+    {
+        return this;
+    }
+
+    void popFront() @property
+    {
+        assert(!empty, "Attempting to pop the front of an empty TakeWhile range");
+        _range.popFront;
+    }
+}
+
+/**
+Lazily produce elements of a range as long as the predicate holds
+for the produced elements.
+
+Params:
+    range = the range to take from while the predicate stands
+    pred = the prediacate used to take the elements
+
+Returns:
+    A `ForwardRange` which contains the taken elements.
+
+See_Also: $(LREF chain) to chain ranges
+ */
+auto takeWhile(alias pred, R)(R range) if (isInputRange!R)
```

Comment:
IIRC, I tried to add `takeWhile` a number of years ago, and he vetoed it along with other stuff like `dropWhile`. Basically, the decision was that we could do what we needed to do with what we had, and you're always going to have problems with folks finding stuff, because they assume one name, whereas the person who designed the function thought of another - and changing the name often just results in a different set of people not finding what they're looking for (a prime example of this would be `strip` vs `trim` - some languages/libraries use one and some the other, and both are valid names, but if the library you're trying to use didn't use the one you're thinking of, you might not find it). And we really can't fix that problem. You try to pick a good name and then try and make sure that your documentation makes it easy to find.

In this case, `until` already does what this PR does - it even has an overload that doesn't take a sentinel. And honestly, the fact that is _does_ have the ability to indicate a sentinel as well as whether it's open on the right makes it more useful than this implementation of `takeWhile`.

Certainly, I see no reason to add this code. It's just doing what `until` already does - only less. At most, it would make sense to create an alias or to make `takeWhile` a wrapper around `until`, but neither would help functionality at all. It's just bikeshedding over the name, and while it might make it easier for some people to find the function, it's also going to result in confusion over the difference between `takeWhile` and `until`. Our policy has generally been to avoid adding aliases to provide additional names.

So, while I think that this is ultimately up to Andrei, I don't think that it makes sense to add `takeWhile` given that we already have `until`, much as I agree that `takeWhile` would likely have been a better name.